lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -33,47 +46,53 @@ def execute(self):
         else:
             groups = self.groups
 
+        opts = self.context.with_option
+        tool_info = {}
+
         if not self.context.name:
-            host_tools = {}
             for group in groups:
-                host_tools[group] = {}
+                tool_info[group] = {}
                 try:
                     for path in self.gen_tools_group_dir(group).glob("*/**"):
-                        if self.context.with_option:
-                            host_tools[group][path.name] = [
-                                (p, (path / p).read_text()) for p in self.tools(path)
-                            ]
-                        else:
-                            host_tools[group][path.name] = [p for p in self.tools(path)]
+                        host = path.name
+                        tool_info[group][host] = {}
+                        for tool in sorted(self.tools(path)):
+                            tool_info[group][host][tool] = sorted((path / tool).read_text().rstrip('\n').split('\n')) if opts else ""
                 except BadToolGroup:
                     self.logger.error("Bad tool group: %s", group)
-                    # 0.69.9 behavior
-                    return 0
-            if host_tools:
-                for k, v in host_tools.items():
-                    for h, t in v.items():
-                        print("%s: %s %s" % (k, h, t))
+                    return 1
+            if tool_info:
+                self.print_results(tool_info, None, self.context.with_option)
         else:
             # List the groups which include this tool
-            group_list = []
+            tool = self.context.name
+            found = False
             for group in groups:
-                tg_dir = self.gen_tools_group_dir(group)
-                if not tg_dir.exists():
-                    self.logger.error("bad or missing tool group %s", group)
-                    continue
+                try:
+                    tg_dir = self.gen_tools_group_dir(group)
+                except BadToolGroup:
+                    self.logger.error("Bad tool group: %s", group)
+                    return 1
 
+                tool_info[group] = {}
                 for path in tg_dir.iterdir():
```

Comment:
I kind of like this idiom (with the small body of the `try` block); please consider using it for the previous call to `get_tools_group_dir()`.

Reply:
Done.